lib/ibmiconnector.js

```diff
@@ -1124,36 +1268,61 @@ class IBMiConnector extends SQLConnector {
+    return this.execute(`BEGIN IF EXISTS (SELECT NAME FROM SYSIBM.TABLES
+      WHERE${this.schema(model) ? ` TABLE_SCHEMA = '${this.schema(model)}'
+      AND` : ''} TABLE_NAME = '${this.table(model)}')
+      THEN DROP TABLE ${this.tableEscaped(model)}; COMMIT; END IF; END`, cb);
   }
 
   ping(callback) {
     const sql = 'SELECT 1 AS PING FROM SYSIBM.SYSDUMMY1';
-    this.executeSQL(sql, callback);
+    this.execute(sql, callback);
+  }
+
+  searchForPropertyInActual(model, propName, actualFields) {
+    let found = false;
+    actualFields.forEach((f) => {
+      if (f.name === this.column(model, propName)) {
+        found = f;
+      }
+    });
+    return found;
   }
 
-  // ///////////////////////////////////////////////////////////////////////////////////////////////
-  // /////////////////////////////////// TRANSACTIONS //////////////////////////////////////////////
-  // ///////////////////////////////////////////////////////////////////////////////////////////////
+  // ///////////////////////////////////////////////////////////////////////////
+  // //////////////////////////// TRANSACTIONS /////////////////////////////////
+  // ///////////////////////////////////////////////////////////////////////////
+
+  // This functionality is standard to all DBMS using ODBC drivers. In
+  // particular, ODBC connections should not use BEGIN_TRANSACTION, ROLLBACK,
+  // or COMMIT statements, and instead should use ODBC functions like
+  // SQLSetConnectAttr to start and SQLEndTran to end.
 
-  // This functionality is standard to all DBMS using ODBC drivers. In particular, ODBC connections
-  // should not use BEGIN_TRANSACTION, ROLLBACK, or COMMIT statements, and instead should use ODBC
-  // functions like SQLSetConnectAttr to start and SQLEndTran to end. Therefore, this functionality
-  // is handled by the loopback-odbc layer.
   /**
    * Begin a new transaction
 
-   * @param {Integer} isolationLevel
+   * @param {Integer} level
    * @param {Function} cb
    */
-  beginTransaction(isolationLevel, callback) {
-    // TODO: Manage isolation level?
-    debug('beginTransaction: isolationLevel: %s', isolationLevel);
+  beginTransaction(level, callback) {
+    debug('beginTransaction: isolationLevel: %s', level);
     this.pool.connect((error1, connection) => {
-      if (error1) { return callback(error1, null); }
-      return connection.beingTransaction((error2) => {
-        if (error2) { return callback(error2, null); }
-        return callback(null, connection);
+      if (error1) {
+        return callback(error1);
+      }
+      connection.setIsolationLevel(this.transactionMap[level], (error2) => {
+        if (error2) {
+          return callback(error2);
+        }
+        return connection.beginTransaction((error3) => {
+          if (error3) {
+            return callback(error3);
+          }
+          const tx = new Transaction(this, connection);
+          tx.txId = uuid.v1();
```

Comment:
It's usually better to use uuid v4, see https://github.com/tc39/proposal-uuid#arent-v1-uuids-better-because-they-are-guaranteed-to-be-unique